AggregateMetrics/AggregateMetrics/AzureWebApp/PerformanceCollectorModule.cs

```diff
@@ -21,14 +24,40 @@ public class PerformanceCollectorModule : ITelemetryModule
                                                                 @"\Processor(_Total)\% Processor Time"
                                                             };
 
-        public IList<PerformanceCounterCollectionRequest> Counters { get; private set; }
+        public List<PerformanceCounterCollectionRequest> Counters { get; private set; }
 
+        /// <summary>
+        /// Initializes the default performance counters.
+        /// </summary>
         public void Initialize(TelemetryConfiguration configuration)
         {
             CounterFactory factory = new CounterFactory();
 
             foreach (string counter in defaultCounters)
-                factory.GetCounter(counter);
+            {
+                try
+                {
+                    ICounterValue c = factory.GetCounter(counter);
+                    configuration.RegisterCounter(c);
+                }
+                catch (ArgumentException e)
+                {
+                    throw e;
```

Comment:
When you catch and then re-throw the exception directly you lose the stack trace up to that point and start over with the new throw.  If, however, you do just "throw;" and don't specify the exception explicitly the stack trace is preserved.  In this case there is nothing happening so the whole catch is pointless.